src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'clear') {
+      for (const prop in state) {
+        if (state.hasOwnProperty(prop)) {
+          delete state[prop];
+        }
+      }
+    }
+
+    if (actions[i].type === 'addProperties') {
+      Object.assign(state, actions[i].extraData);
+    }
+
+    if (actions[i].type === 'removeProperties') {
+      for (let j = 0; j < actions[i].keysToRemove.length; j++) {
+        delete state[actions[i].keysToRemove[j]];
+      }
+    }
```

Comment:
use switch..case to do smth based on action.type

Reply:
Done. The solution was updated with suggested changes.